### DISCUSSION TITLE
Access what date def next(self): is currently on?

### DISCUSSION THREAD

**Discussion by bioluminesceme** [kernc/backtesting.py#663] [Q&A] [answered]
Is there an easy way to access what date def next is currently on?

I tried searching the docs but only found OHLCV in self.data 
I think the date should be in there too somewhere, since that's the first column?

    def next(self):
        current_close = self.data.Close
        currentdate = self.data.?

Thanks!

**Comment by kernc** [comment, marked as answer]
See [`Strategy.data.index`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.data).
```py
    def next(self):
        current_bar_idx = len(self.data) - 1
        current_date = self.data.index[-1]
```

**Reply by bioluminesceme** [reply]
Thank you!! ( I tried strategy.data.i that I read in another question here  - I was so close! )
